### PR TITLE
Include error in codestatus response

### PR DESCRIPTION
**Pull request by whaught**
## Proposed Changes

* Add the error fields to the CheckCodeStatusResponse struct
* Minor logging statement and a nil check

**Release Note**

```release-note
Include errors in CheckCodeStatusResponse
```

### CODE REVIEW

File: pkg/controller/issueapi/codestatus.go [hunk 1]
```diff
@@ -68,13 +68,16 @@ func (c *Controller) HandleCheckCodeStatus() http.Handler {
...
 		if code.UUID == "" { // if no row is found, code will not be populated
 			logger.Errorw("failed to check otp code status", "error", "code not found")
 			c.h.RenderJSON(w, http.StatusNotFound, api.Errorf("failed to check otp code status"))
 			return
 		}
 
-		if code.IssuingUser.Email != user.Email && !user.CanAdminRealm(realm.ID) {
+		// The current user must have issued the code or be a realm admin.
+		if !(code.IssuingUser != nil && code.IssuingUser.Email == user.Email || user.CanAdminRealm(realm.ID)) {
 			logger.Errorw("failed to check otp code status", "error", "user email does not match issuing user")
```
Comment (mikehelmick) [inline pkg/controller/issueapi/codestatus.go:81]: can you assign ErrorCodes to these?

Comment (whaught) [reply pkg/controller/issueapi/codestatus.go:84]: yes. Done

File: pkg/api/api.go [hunk 2]
```diff
@@ -36,6 +36,10 @@ const (
 	ErrVerifyCodeInvalid = "code_invalid"
 	// ErrVerifyCodeExpired indicates the code provided is known to the server, but expired.
 	ErrVerifyCodeExpired = "code_expired"
+	// ErrVerifyCodeNotFound indicates the code does not exist on the server/realm.
+	ErrVerifyCodeNotFound = "code_not_found"
```
Comment (mikehelmick) [inline pkg/api/api.go:40]: can you organize these by the API they belong to - there aren't part of the

**Pull request merged**